ldap/servers/slapd/back-ldbm/db-bdb/bdb_bdbreader_glue.c

```diff
@@ -471,7 +471,7 @@ int db_cursor(DB *db, DB_TXN *txnid, DBC **cursorp, u_int32_t flags)
     dbc->impl = bdbreader_cur_open(db->impl);
 
     slapi_log_err(SLAPI_LOG_INFO, "bdb_ro", "%s: dbc=%p dbc->impl=%p\n", __FUNCTION__, dbc, dbc->impl);
-    return (db->impl == NULL) ? DB_OSERROR : DB_SUCCESS;
+    return (dbc->impl == NULL) ? DB_OSERROR : DB_SUCCESS;
```

Comment:
I think this is correct @progier389   You said about moving the NULL check up, but I think it supposed to be "dbc" not  "db"